src/common/ipc/unix_socket_client.hpp

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+//
+
+#ifndef OCVSMD_COMMON_IPC_UNIX_SOCKET_CLIENT_HPP_INCLUDED
+#define OCVSMD_COMMON_IPC_UNIX_SOCKET_CLIENT_HPP_INCLUDED
+
+#include <string>
+
+namespace ocvsmd
+{
+namespace common
+{
+namespace ipc
+{
+
+class UnixSocketClient final
+{
+public:
+    explicit UnixSocketClient(std::string socket_path);
+
+    UnixSocketClient(UnixSocketClient&&)                 = delete;
+    UnixSocketClient(const UnixSocketClient&)            = delete;
+    UnixSocketClient& operator=(UnixSocketClient&&)      = delete;
+    UnixSocketClient& operator=(const UnixSocketClient&) = delete;
+
+    ~UnixSocketClient();
+
+    bool connect_to_server();
+    void send_message(const std::string& message) const;
```

Comment:
can we also support string_view?

Reply:
this `send_message` was just to prove send and receive. real communication api of course will be different. probably passing back and forth some serialized structures.